workspaces/adventure-pack/goodies/python3/src/transpose_matrix/test.py

```diff
@@ -0,0 +1,51 @@
+from . import *
+import pytest
+
+
+def test_improper_input() -> None:
+    with pytest.raises(ValueError):
+        assert len(transpose_matrix([]))
+
+    with pytest.raises(ValueError):
+        assert len(transpose_matrix([[], []]))
+
+    with pytest.raises(ValueError):
+        assert len(transpose_matrix([[0], [0, 1]]))
+
+
+def assert_matrices_equal(list1, list2):
+    assert len(list1) == len(list2)
+    for row in range(len(list1)):
+        assert list1[row] == list2[row]
+
+
+def test_nonsymetric_transpose() -> None:
+    input_1 = [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]]
+    input_2 = [
+        ["0", "1", "2"],
+        ["3", "4", "5"],
+        ["6", "7", "8"],
+        ["9", "10", "11"],
+    ]
```

Comment:
I think rather than putting each input first, then each expected, then doing all the asserts, it reads a bit better if we do input1/expected1/assert, input2/expected2/assert, etc.